From 902f93c1b4da5bc3d8e07a67594f6a68077eae14 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Wed, 4 Mar 2026 23:39:29 +0000
Subject: [PATCH 1/1] Replace hardcoded Google captive portal fallback URL

The CaptivePortalLoginActivity has a hardcoded fallback URL pointing to
connectivitycheck.gstatic.com that is used when no URL is provided via
intent extras. Unlike the NetworkMonitor captive portal URLs, this
fallback is not governed by the ConnChecksSetting dual-mode switch and
will unconditionally contact Google. Replace it with the GrapheneOS
connectivity check endpoint.

Change-Id: I4d0b7c199f84f60fa17db3f877a3e8c35b5e31c5
---
 .../android/captiveportallogin/CaptivePortalLoginActivity.java  | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/captiveportallogin/CaptivePortalLoginActivity.java b/src/com/android/captiveportallogin/CaptivePortalLoginActivity.java
index 31b6729..1b302fb 100755
--- a/src/com/android/captiveportallogin/CaptivePortalLoginActivity.java
+++ b/src/com/android/captiveportallogin/CaptivePortalLoginActivity.java
@@ -172,7 +172,7 @@ public class CaptivePortalLoginActivity extends Activity {
     private static final int SOCKET_TIMEOUT_MS = 10000;
     public static final String HTTP_LOCATION_HEADER_NAME = "Location";
     private static final String DEFAULT_CAPTIVE_PORTAL_HTTP_URL =
-            "http://connectivitycheck.gstatic.com/generate_204";
+            "http://connectivitycheck.grapheneos.network/generate_204";
     // This should match the FileProvider authority specified in the app manifest.
     private static final String FILE_PROVIDER_AUTHORITY =
             "com.android.captiveportallogin.fileprovider";
-- 
2.53.0

